extensions/arc/src/models/controllerModel.ts

```diff
@@ -50,33 +51,53 @@ export class ControllerModel {
 		this._onInfoUpdated.fire(this._info);
 	}
 
+	public get azdataAdditionalEnvVars(): { [key: string]: string } {
+		return {
+			'KUBECONFIG': this.info.kubeConfigFilePath,
+			'KUBECTL_CONTEXT': this.info.kubeClusterContext
+		};
+	}
+
 	/**
 	 * Calls azdata login to set the context to this controller
 	 * @param promptReconnect
 	 */
 	public async azdataLogin(promptReconnect: boolean = false): Promise<void> {
-		// We haven't gotten our password yet or we want to prompt for a reconnect
-		if (!this._password || promptReconnect) {
+		let promptForValidClusterContext: boolean = false;
+		try {
+			const contexts = await getKubeConfigClusterContexts(this.info.kubeConfigFilePath);
+			promptForValidClusterContext = getCurrentClusterContext(contexts, this.info.kubeClusterContext, true) === this.info.kubeClusterContext;
+		} catch (error) {
+			const response = await vscode.window.showErrorMessage(loc.clusterContextConfigNoLongerValid(this.info.kubeConfigFilePath, this.info.kubeClusterContext, error), ...[loc.yes, loc.no]);
```

Comment:
Spreading the array is unnecessary, just pass in the values directly 

`, loc.yes, loc.no);`

Reply:
done